File: MathCalculations/Kotlin/SieveOfEratosthenes/SieveOfEratosthenes.kt
fun sieveOfEratosthenes(limit: Int): List<Int> {
    
    val isPrime = BooleanArray(limit + 1) { true }
    val primes = mutableListOf<Int>()

    for (num in 2..limit) {
        if (isPrime[num]) {
            primes.add(num)

            var multiple = 2 * num
            while (multiple <= limit) {
                isPrime[multiple] = false
                multiple += num
            }
        }
    }

    return primes
}

fun main() {
    val limit = 1000000 
    val stopwatch = System.currentTimeMillis()
    sieveOfEratosthenes(limit)
    var elapsedTime = System.currentTimeMillis() - stopwatch
    var formattedTime = String.format("%.2f", elapsedTime.toDouble())
    formattedTime = formattedTime.replace(',', '.')
    println("$formattedTime")
    //println("Prime numbers up to $limit: $primeNumbers")
}

File: MathCalculations/Dart/SieveOfEratosthenes/SieveOfEratosthenes.dart
List<int> sieveOfEratosthenes(int limit) {
  List<bool> isPrime = List<bool>.filled(limit + 1, true);
  List<int> primes = [];

  for (int num = 2; num <= limit; num++) {
    if (isPrime[num]) {
      primes.add(num);

      int multiple = 2 * num;
      while (multiple <= limit) {
        isPrime[multiple] = false;
        multiple += num;
      }
    }
  }

  return primes;
}

void main() {
  int limit = 1000000; 
  final stopwatch = Stopwatch()..start();
  List<int> primeNumbers = sieveOfEratosthenes(limit);
  stopwatch.stop();
  double elapsedTime = stopwatch.elapsedMilliseconds.toDouble();
  String formattedTime = elapsedTime.toStringAsFixed(2);
  print(formattedTime);
  //print("Prime numbers up to $limit: $primeNumbers");
}
